core/src/main/java/org/infinispan/registry/InternalCacheRegistry.java

```diff
@@ -18,17 +18,45 @@
 public interface InternalCacheRegistry {
    enum Flag {
       EXCLUSIVE, // means that the cache must be declared only once
-      USER,  // means that this cache is visible to users
+      USER, // means that this cache is visible to users
       PERSISTENT, // means the cache should be made persistent across restarts if global state persistence is enabled
    }
 
+   /**
+    * Registers an internal cache. The cache will be marked as private and volatile
+    *
+    * @param name
+    *           The name of the cache
+    * @param configuration
+    *           The configuration for the cache
+    */
    void registerInternalCache(String name, Configuration configuration);
 
+   /**
+    * Registers an internal cache with the specified flags.
+    *
+    * @param name
+    *           The name of the cache
+    * @param configuration
+    *           The configuration for the cache
+    * @param flags
+    *           The flags which determine the behaviour of the cache. See {@link Flag}
+    */
    void registerInternalCache(String name, Configuration configuration, EnumSet<Flag> flags);
 
+   /**
+    * Returns whether the cache is internal, i.e. it has been registered using the
+    * {@link #registerInternalCache(String, Configuration)} method
+    */
    boolean isInternalCache(String name);
 
+   /**
+    * Retrieves the names of all the internal caches
+    */
    Set<String> getInternalCacheNames();
 
-   void filterInternalCaches(Set<String> names);
+   /**
+    * Removes the internal caches from the specified set of cache names
+    */
+   void filterPrivateCaches(Set<String> names);
```

Comment:
The javadoc is wrong and the actual implementation was filtering the private caches :) I'll fix this